src/controllers/controller.bar.js

```diff
@@ -327,9 +327,11 @@ module.exports = DatasetController.extend({
 					imeta.controller.getValueScaleId() === scale.id &&
 					chart.isDatasetVisible(i)) {
 
-					ivalue = scale.getRightValue(datasets[i].data[index]);
+					ivalue = +scale.getRightValue(datasets[i].data[index]);
 					if ((value < 0 && ivalue < 0) || (value >= 0 && ivalue > 0)) {
 						start += ivalue;
+					} else if (minBarLength && value === 0 && ivalue === 0) {
+						start = scale.getValueForPixel(scale.getPixelForValue(start) + minBarLength);
```

Comment:
If a stack contains many zero values, `start` won't point to the right value and the size of bars for non-zero values may go wrong. Since `minBarLength` has been introduced just to indicate if a bar exists at that index, how about not adding `minBarLength` here and keeping the code as is? (If a stack contains many zero values, bars with `minBarLength` are drawn at the same position many times, but it at least tells that a bar exists there.)

By the way, https://github.com/chartjs/Chart.js/blob/994497c9e24e00094a4737a66ce6caaf0be38c9b/src/controllers/controller.bar.js#L342 is wrong, so can you change it to `size = head - base;`?